xdsl/dialects/riscv.py

```diff
@@ -331,6 +334,73 @@ class RISCVOp(Operation, ABC):
     def assembly_line(self) -> str | None:
         raise NotImplementedError()
 
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        operands = cls.parse_operands(parser)
+        successors = cls.parse_successors(parser)
+        regions = cls.parse_regions(parser)
+        attributes = cls.parse_attributes(parser)
+        parser.parse_punctuation(":")
+        result_types = cls.parse_function_type(parser).outputs.data
+        return cls.create(
+            operands=operands,
+            result_types=result_types,
+            attributes=attributes,
+            successors=successors,
+            regions=regions,
+        )
+
+    @classmethod
+    def parse_operands(cls, parser: Parser) -> Sequence[SSAValue]:
+        return parser.parse_comma_separated_list(
+            parser.Delimiter.PAREN, parser.parse_operand
+        )
+
+    @classmethod
+    def parse_successors(cls, parser: Parser) -> Sequence[Block]:
+        return parser.parse_optional_successors() or []
+
+    @classmethod
+    def parse_function_type(cls, parser: Parser) -> FunctionType:
+        return parser.parse_function_type()
+
+    @classmethod
+    def parse_attributes(cls, parser: Parser) -> Mapping[str, Attribute]:
+        return parser.parse_optional_attr_dict()
+
+    @classmethod
+    def parse_regions(cls, parser: Parser) -> Sequence[Region]:
+        if parser.parse_optional_punctuation("(") is not None:
+            regions = parser.parse_comma_separated_list(
+                parser.Delimiter.NONE, parser.parse_region
+            )
+            parser.parse_punctuation(")")
+            return regions
+        return []
```

Comment:
Similarly unused

Reply:
Just to double check, did you test this?

Reply:
Ah, I think you are right, it's not unused! Good catch!

Reply:
This code should already exist in the parser, no?

Reply:
There is currently no `parse_optional_regions` (there is `parse_region` though).

Reply:
This one will be available promptly. We can then again drop the `parse_regions` function.

Reply:
There is no parse_region_list, which you can use here.

Reply:
Now using the `parser.parse_region_list`. Mark as resolved.